clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1484,6 +1495,16 @@ void KernelObjVisitor::visitArray(const CXXRecordDecl *Owner, FieldDecl *Field,
 
   if (Field->hasAttr<SYCLRequiresDecompositionAttr>()) {
     visitComplexArray(Owner, Field, ArrayTy, Handlers...);
+  } else if (AnyTrue<HandlerTys::VisitInsideSimpleContainersWithPointer...>::
+                 Value) {
+    assert(!Field->hasAttr<SYCLGenerateNewTypeAttr>() &&
+           "Arrays should trigger decomposition");
+    // We are currently in PointerHandler visitor, which implies this is a
+    // 'simple' array i.e. one that does not include special types or pointers.
+    // Array of pointers/ array of type containing pointers will be handled in
+    // a follow-up PR. Currently, they continue to trigger decomposition, and
+    // will be handled in 'if' statement above.
```

Comment:
Should we add FIXME comment here like line no:1873-1875?

Reply:
I can if you want me to. It doesn't matter to me one way or the other. I think I just left it off here because I saw Line1504 as a part of para which starts on L1502

Reply:
It is fine to me. You can change it if there is other review comments otherwise it is not necessary. Thanks Elizabeth.

Reply:
Thanks for reviewing!